HW3_spark_smythe.py

```diff
@@ -1,156 +0,0 @@
-from split_column_transform import SplitColumnTransform
-
-
-# copy from class notes for custom trasnform
-class SplitColumnTransform(
-    Transformer,
-    HasInputCols,
-    HasOutputCol,
-    DefaultParamsReadable,
-    DefaultParamsWritable,
-):
-    @keyword_only
-    def __init__(self, inputCols=None, outputCol=None):
-        super(SplitColumnTransform, self).__init__()
-        kwargs = self._input_kwargs
-        self.setParams(**kwargs)
-        return
-
-    @keyword_only
-    def setParams(self, inputCols=None, outputCol=None):
-        kwargs = self._input_kwargs
-        return self._set(**kwargs)
-
-    def _transform(self, dataset):
-        input_cols = self.getInputCols()
-        output_col = self.getOutputCol()
-
-        # Generate a list of columns to append
-        concat_list = []
-        for count, column in enumerate(input_cols):
-            column_func = when(col(column).isNull(), lit("")).otherwise(col(column))
-            concat_list.append(column_func)
-            if count < len(input_cols) - 1:
-                concat_list.append(lit(" "))
-        dataset = dataset.withColumn(
-            output_col, split(concat(*concat_list), pattern=" ")
-        )
-        return dataset
-
-
-def main():
-    # Setup Spark
-    spark = SparkSession.builder.master("local[*]").getOrCreate()
-
-    # set up sql defaults
-    database = "baseball"
-    port = "3306"
-    user = "root"
-    password = "x11desktop"  # pragma: allowlist secret
-
-    # get table and make batter df
-    bb_df = (
-        spark.read.format("jdbc")
-        .options(
-            url=f"jdbc:mysql://localhost:{port}/{database}",
-            driver="com.mysql.cj.jdbc.Driver",
-            dbtable="batter_counts",
-            user=user,
-            password=password,
-            inferSchema="true",
-            header="true",
-        )
-        .load()
-    )
-
-    bb_df.select("batter", "game_id", "atBat", "Hit").show()
-    bb_df.createOrReplaceTempView("batter")
-    bb_df.persist(StorageLevel.DISK_ONLY)
-
-    # get table and make game df
-    gd_df = (
-        spark.read.format("jdbc")
-        .options(
-            url=f"jdbc:mysql://localhost:{port}/{database}",
-            driver="com.mysql.cj.jdbc.Driver",
-            dbtable="game",
-            user=user,
-            password=password,
-            inferSchema="true",
-            header="true",
-        )
-        .load()
-    )
-
-    gd_df.select("game_id", "local_date").show()
-    gd_df.createOrReplaceTempView("game")
-    gd_df.persist(StorageLevel.DISK_ONLY)
-
-    bat_avg = spark.sql(
-        """
-        select
-            batter
-            , atBat
-            , Hit 
-            , (Hit/nullif(atBat,0)) as game_bat_avg
-        from batter
-        """
-    )
-    bat_avg.show()
-    # make pipeline
-
-    split_column_transform = SplitColumnTransform(
-        inputCols=["batter", "game_id", "atBat", "Hit"], outputCol="categorical"

```

Comment:
All of these input columns are actually integers in SQL so I believe none of them have to be transformed. From what I understand you want to use 'SplitColumnTransform' when you have categorical data such as 'sex' and typically they are words but not always. The set up is correct though!

Reply:
ok. Thank you